editor/code_editor.cpp

```diff
@@ -2087,13 +2044,42 @@ CodeTextEditor::CodeTextEditor() {
 	warning_button->set_tooltip_text(TTR("Warnings"));
 	set_warning_count(0);
 
+	status_bar->add_child(memnew(VSeparator));
+
+	// Zoom
+	zoom_button = memnew(MenuButton);
```

Comment:
I remembered the other reason I chose a MenuButton is because the text isn't always a reflection of the selected text, and in fact, there isn't even always a selection. So constantly managing the text override and the selection index when there's a custom zoom (which is most of the time), was less straightforward than just using a MenuButton, which is what a lot of things in the editor's bars use.

Unless there's another good reason not to use a MenuButton I'm gonna mark this one as resolved.